api/controller/triggers.go

```diff
@@ -15,7 +15,8 @@ import (
 
 const pageSizeUnlimited int64 = -1
 
-var idValidationPattern = regexp.MustCompile(`^[a-zA-Z0-9\-_]+$`)
+// Only symbols allowed by rfc3986 2.3
+var idValidationPattern = regexp.MustCompile(`^[ \p{L}0-9\.\-\|_~:]+$`)
```

Comment:
Yes, these symbols are not mentioned in 2.3. I should exclude the mentioan of RFC3986 at all, becaule actually all ids can be accessible if properly percent-encoded. Thus the question in not whether the protocols allow them, but whether the frontend can manage to succesfully process them.

By the way, these symols are allowed here because they are actually used in our production database and are processed correctly.

Reply:
Deleted the comment entirely

Reply:
> I should exclude the mention of RFC3986 at all

Ok

>  becaule actually all ids can be accessible if properly percent-encoded

When users use Ansible role to create triggers, I think they don't use percents, so it is has no sense, nobody will encode string manually. 


> By the way, these symols are allowed here because they are actually used in our production database and are processed correctly.

It is not well, but ok...

Reply:
I have to mention that task of validation trigger ID was created, when we discover trigger with `#` in ID. That trigger can't be viewed by a web browser. 

So what is the goal to make pattern more weak (I mean continue allow use space and `:`, `|`). Is it strict validation somehow affects existing triggers?

Reply:
It won't affect existing triggers. But some triggers are created automatically, which may result in unexpected loss of data. I'd prefer to play it safe for now. We will find out how these strange triggers are created and than (hopefully) enforce more strict validation

Reply:
Oh yes, if we do more strict validation, then existing triggers will not be recreated. I agree, thanks